src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/DependencyNode.cs

```diff
@@ -126,6 +127,32 @@ internal DependencyNode()
         public string Caption { get; set; }
 
         /// <summary>
+        /// Actual formal name of the node. In most cases it will be equal to Caption,
+        /// however in some cases caption would have name + version or something else.
+        /// So to allow tree provider store formal name we need this property too. 
+        /// It is a hack now and we should make it public and move to IDependencyNode when
+        /// we get a chance and it is safe to do a breaking change.
```

Comment:
i agree. I opened a bug for RTW and if it is approved we coudl do that - it is not hard to implement, but logistics ... we need to synchronize insertions for core and web. It also can be done , just need an agreement :). Btw David Kean wanted also to rename some public Async methods , we could combine 2 changes.

---
In reply to: [95038283](https://github.com/dotnet/roslyn-project-system/pull/1091#discussion_r95038283) [](ancestors = 95038283)